pkg/handlers/config.go

```diff
@@ -205,14 +205,24 @@ func UpdateAppConfig(w http.ResponseWriter, r *http.Request) {
 			return
 		}
 	} else {
-		err := app.UpdateConfigValuesInDB(archiveDir, foundApp.ID, int64(updateAppConfigRequest.Sequence))
-		if err != nil {
+		if err := app.UpdateConfigValuesInDB(archiveDir, foundApp.ID, int64(updateAppConfigRequest.Sequence)); err != nil {
 			logger.Error(err)
 			updateAppConfigResponse.Error = "failed to update config values in db"
 			JSON(w, 500, updateAppConfigResponse)
 			return
 		}
 
+		downstreamStatus := "pending"
+		if kotsKinds.Preflight != nil {
+			downstreamStatus = "pending_preflight"
+		}
+		if err := app.UpdateDownstreamVersionStatus(foundApp.ID, int64(updateAppConfigRequest.Sequence), downstreamStatus, ""); err != nil {
```

Comment:
How would you feel about making this clearer? UpdateDownsteamVersionStatus doesn't seem to be very identifiable to describe what it does.

I'm wondering if 2 functions: 

app.SetDownstreamVersionReady(appID, sequence)
app.SetDownstreamVersionPendingPreflights(appID, sequence)

And then the logic is more separated?

Reply:
I feel this will cause more code duplication in the future if we decide to add more statuses to the downstream version. We already have `pending`, `pending_preflight`, `pending_config` and `deployed`.

Reply:
Is duplication of this function bad when it's more clear? How would i know to set the status to pending_preflight and status info to "" to wait for preflights? Since there are multiple methods and paths that could trigger this status, having the status fields spread around and the responsibility of the caller to minimize code duplication feels like a path to creating bugs when one caller sends it in differently

Reply:
That's a good point. I will update.